### ISSUE TITLE
Unify battery charging ports

### ISSUE THREAD

**Issue by ducky64** [BerkeleyHCI/PolymorphicBlocks#330]
Currently, batteries have a pwr (output) and chg (input) Voltage port, even though both are on the same copper net. Can be unintuitive and lead to weirdness

Needs more thought on how to handle this. Perhaps a bidirectional port? Perhaps some concept of chargers (regulated voltage / current devices) instead of the constant-voltage source?

Some example topologies it should support:
- (battery) - (charger + sinks)
  - Note: this has a source (charger) + bidir (battery), which in general may not be safe.
- (battery) - (sinks)
- (battery) - (reverse protection circuit - can be charging aware) - (charger + sinks)

Some example topologies that might error:
- (multiple batteries in parallel) - (anything else)
- (battery) - (non-charging-capable reverse protection circuit, eg simple PMOS device) - (charger + sinks)
  - more of a stretch goal to detect this and error
- (charger) - (sinks)
  - Technically can be fine, where the charger is functionally a current-limited supply, but maybe unusual
